packages/integration-testsuite/src/httpServerTests.ts

```diff
@@ -2060,6 +2061,31 @@ export function defineIntegrationTestSuiteHttpServerTests(
           expect(res.body.errors[0].message).toEqual(expected);
         });
       });
+
+      it('uses stringifyResult parameter', async () => {
+        const app = await createApp({
+          schema,
+          stringifyResult: (value: FormattedExecutionResult) => {
+            let result = JSON.stringify(value, null, 1); // stringify, with line-breaks and indents
+            result = result.replace('it works', 'stringifyResults works!'); // replace text with something custom
+            return result;
+          },
+        });
+        const expected = {
+          testString: 'stringifyResults works!',
+        };
+        const query = {
+          query: 'query test{ testString }',
+        };
+        const req = request(app)
+          .get('/')
+          .set('apollo-require-preflight', 't')
+          .query(query);
+        return req.then((res) => {
+          expect(res.status).toEqual(200);
+          expect(res.body.data).toEqual(expected);
+        });
+      });
```

Comment:
I think this is specific to Apollo Server and doesn't necessarily need to be tested for all integrations. I think this might belong in `ApolloServer.test.ts` instead under a `describe('configuration options', ...)` block.

There seems to be a couple layers in testing this. Your `replace` is clearly working as expected but the additional `JSON.stringify` formatting arguments are of no consequence to this test (i.e. I can remove them and the test still passes) so I'd rather see a test that validates the actual string output (probably with `toMatchInlineSnapshot`) or something similar.

Sidenote: `res.body.data` is a parsed object, so it seems like `res.text` would be more appropriate for validating the formatting changes.